metadata-ingestion/src/datahub/ingestion/source/aws/sagemaker.py

```diff
@@ -92,6 +92,9 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         # extract jobs if specified
         if self.source_config.extract_jobs is not False:
             job_processor = JobProcessor(
+                config=self.source_config
```

Comment:
I am not a big fan of having 2nd sources of `SageMakerClient`. I understand that we want to trigger `get_session()` in case we are using roles. It seems though that in other cases `get_session()` would simply return client using the existing credentials, which will lead to having a proper client object.
Another change which I would suggest is to not pass whole `self.source_config` object to the `JobProcessor` - I understand we want to trigger the `get_session()`, but this could be achieved, for example, by passing a function instead of a value (just like implemented in `SageMakerConfig`). If `@property` decorator is standing in the way here we could implement a function in the config and pass it (not the value it returns but rather reference to the function to be executed everytime the client is needed instead).
These combined changes would make this PR much smaller as well.

Reply:
Fixed can you please re review again Thanks !!!